flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -90,6 +91,10 @@ public static CheckpointedInputGate createCheckpointedInputGate(
 			inputGate1.getNumberOfInputChannels() + inputGate2.getNumberOfInputChannels(),
 			taskName,
 			toNotifyOnCheckpoint);
+
+		TaskIOMetricGroup taskIOMetricGroup = toNotifyOnCheckpoint.getEnvironment().getMetricGroup().getIOMetricGroup();
```

Comment:
nit: pass through constructor? Getting metric group from a variable named `toNotifyOnCheckpoint` looks a bit strange?

Reply:
I agree with the point, and the current way is for avoiding the more changes of existing tests. Let me try that.

Reply:
Fixed